canonical-web-and-design/maas-ui
# fix(ui): fix route errors and add tests

## Done

- Fix an issue where routes were causing hook errors.
- Add tests for all routes.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit settings, preferences and /r/networks?by=fabric.
- They should all load and there shouldn't be any errors about invalid hooks.

## Fixes

Fixes: #3537.
